src/unit.rs

```diff
@@ -277,14 +284,20 @@ fn connect_inner(
     }
     let retryable = unit.is_retryable(&body);
 
+    let mut stream = stream::DeadlineStream::new(stream, unit.deadline);
+
+    if expect_100_continue {
+        let response = Response::read_response_head(&mut stream, unit)?;
+        if response.status() != 100 {
+            return Err(Error::Status(response.status(), response));
```

Comment:
This code is broken, as it returns a "response" with the stream set to PendingReader::BeforeBodyStart. This means that we will panic if someone tries to read the response body.